### ISSUE TITLE
test failure on Fedora rawhide with python3.5

### ISSUE THREAD

**Issue by keszybz** [joblib/joblib#278]
This appears to be different than other failures with python3.5 that were already fixed...

http://koji.fedoraproject.org/koji/taskinfo?taskID=11968168
with python3-numpy-1.10.2-0.1rc1.fc24 and python3-3.5.0-5.fc24.

```
======================================================================
FAIL: joblib.test.test_hashing.test_hashes_stay_the_same_with_numpy_objects('94fb1c385e1e955b74ead88ffaf6dcc8', '6028954a4c71750d74e66414bbab227b')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
AssertionError: '94fb1c385e1e955b74ead88ffaf6dcc8' != '6028954a4c71750d74e66414bbab227b'
======================================================================
FAIL: joblib.test.test_hashing.test_hashes_stay_the_same_with_numpy_objects('aaafcb9154f904ad2d387312af7ca907', 'd32171e5bcf0e60034a7923bfea59e08')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
AssertionError: 'aaafcb9154f904ad2d387312af7ca907' != 'd32171e5bcf0e60034a7923bfea59e08'
======================================================================
FAIL: joblib.test.test_hashing.test_hashes_stay_the_same_with_numpy_objects('a271389ac1271a618dbd3e10b688a3c3', '7989a9bbb5b7334efeda618843ab3ecc')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
AssertionError: 'a271389ac1271a618dbd3e10b688a3c3' != '7989a9bbb5b7334efeda618843ab3ecc'
----------------------------------------------------------------------
Ran 626 tests in 11.397s
FAILED (SKIP=1, failures=3)
```

**Comment by sergiopasra** [comment]
This is not a Python 3.5 error. It happens in Python 2.7 also but only in i386

**Comment by sergiopasra** [comment]
The failures happen with release joblib-0.9.3 and with master

**Comment by lesteve** [comment]
I'll try to see if I can reproduce. As a sanity check, you don't happen to have the test results with numpy 1.10.1 by any chance ?

**Comment by ignatenkobrain** [comment]
@lesteve python2-numpy-1.10.2-0.1rc1.fc24.x86_64

**Comment by sergiopasra** [comment]
@lesteve I have tested this with a virtualenv installing everything with pip. numpy is 1.10.1
To reproduce you have to test in a i386 machine. x86_64 works fine

**Comment by lesteve** [comment]
I can reproduce, I'll look into it, thanks for the bug report!

**Comment by lesteve** [comment]
> I can reproduce, I'll look into it, thanks for the bug report!

Just for completeness I am only seeing the test failures on Python 3.5 with a 32bit python.

**Comment by lesteve** [comment]
So the short story on this one: the arrays produced by np.random.randn are not bit-identical on 32bit and 64bit. Not sure why because both of them are float64 arrays:

```python
import numpy as np

rng = np.random.RandomState(0)
arr = rng.randn(3)
print(repr(arr.tobytes()))
```

Here is the outputs with the difference highlighted:
```

32bit:
'\xa5\xcf\xe5\xf3\x8e9\xfc?R\x89\xcc\xfa,\x9c\xd9?\xad\xc9"R\xd2Q\xef?'

64bit:
'\xa5\xcf\xe5\xf3\x8e9\xfc?S\x89\xcc\xfa,\x9c\xd9?\xab\xc9"R\xd2Q\xef?'
                           ^                         ^
                           |                         |
```

As a side-note, this problem doesn't seem to show up on Windows since we are testing both 32bit and 64bit on AppVeyor.

**Comment by lesteve** [comment]
Should be fixed in 002d6f0.
